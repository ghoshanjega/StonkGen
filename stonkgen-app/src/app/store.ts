import { configureStore, ThunkAction, Action,getDefaultMiddleware } from '@reduxjs/toolkit';
import authenticationReducer from '../features/authentication/AuthenticationSlice';
import stocksListReducer from '../features/stocksList/StocksListSlice';
import cartReducer from '../features/cart/CartSlice';

export const store = configureStore({
  reducer: {
    authentication : authenticationReducer,
    stocksList: stocksListReducer,
    cart: cartReducer
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false
}),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
