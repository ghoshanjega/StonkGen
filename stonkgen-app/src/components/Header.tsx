import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link, useHistory } from "react-router-dom";
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  NavLink,
} from "reactstrap";
import { logoutSuccess, selectUser } from "../features/authentication/AuthenticationSlice";
import { Brand } from "./Brand";

export const Header = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const user = useSelector(selectUser);

  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  const handleLogout = () => {
    dispatch(logoutSuccess());
  };
  return (
    <div>
      <Navbar color="dark" dark expand="md">
        <div className="container">
          <NavbarBrand href="/">
            <Brand />
          </NavbarBrand>
          <NavbarToggler onClick={toggle} />
          <Collapse isOpen={isOpen} navbar>
            <Nav className="mr-auto" navbar>
              <button className="btn btn-dark" data-test="home" onClick={() => history.push("/")}>
                Home
              </button>
              <NavItem>
                <button className="btn btn-dark" data-test="orders" onClick={() => history.push("/orders")}>
                  Orders
                </button>
              </NavItem>
              <UncontrolledDropdown nav inNavbar>
                <DropdownToggle nav caret>
                  {user?.fullName}
                </DropdownToggle>
                <DropdownMenu right>
                  {user?.admin && <DropdownItem onClick={() => history.push("/admin")}>Admin</DropdownItem>}
                  <DropdownItem onClick={() => history.push("/order-history")}>History</DropdownItem>
                  <DropdownItem divider />
                  <DropdownItem onClick={() => handleLogout()}>Logout</DropdownItem>
                </DropdownMenu>
              </UncontrolledDropdown>
            </Nav>
            {/* <NavbarText>Simple Text</NavbarText> */}
          </Collapse>
        </div>
      </Navbar>
    </div>
  );
};
