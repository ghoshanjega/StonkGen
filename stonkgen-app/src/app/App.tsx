import React, { useContext } from "react";
import { useSelector } from "react-redux";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { Footer } from "../components/Footer";
import { Header } from "../components/Header";
import { AdminPage } from "../features/admin/AdminPage";

import { AuthenticationPage } from "../features/authentication/AuthenticationPage";
import { selectUser } from "../features/authentication/AuthenticationSlice";
import { CartHistoryPage } from "../features/cart-history/CartHistoryPage";
import { CartPage } from "../features/cart/CartPage";
import { StocksListPage } from "../features/stocksList/StocksListPage";

import "./App.css";

function App() {
  const user = useSelector(selectUser);
  if (!user) {
    return (
      <div>
        <AuthenticationPage />
      </div>
    );
  }
  return (
    <div className="wrapper">
      <BrowserRouter>
        <Header />
        <Switch>
          <Route path="/orders">
            <CartPage />
          </Route>
          <Route path="/order-history">
            <CartHistoryPage />
          </Route>
          <Route path="/admin">
            <AdminPage />
          </Route>
          <Route path="/login">
            <AuthenticationPage />
          </Route>
          <Route path="/">
            <StocksListPage />
          </Route>
        </Switch>
      </BrowserRouter>
      <Footer />
    </div>
  );
}

export default App;
