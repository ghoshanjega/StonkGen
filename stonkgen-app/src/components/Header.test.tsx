import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { store } from "../app/store";

import { Header } from './Header'

describe("Header Component", () => {
    let component: renderer.ReactTestRenderer;
    beforeEach(() => {
      store.dispatch = jest.fn();
  
      component = renderer.create(
        <Provider store={store}>
          <Header />
        </Provider>
      );
    });

    it("should render the header snapshot", () => {
        expect(component).toMatchSnapshot();
      });

      it("renders home button", () => {
        expect(component.root.findByProps({"data-test":"home"}).children).toEqual(["Home"]);
      });

      it("renders orders button", () => {
        expect(component.root.findByProps({"data-test":"orders"}).children).toEqual(["Orders"]);
      });
})