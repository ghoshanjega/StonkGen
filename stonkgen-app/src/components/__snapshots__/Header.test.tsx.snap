// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render the header snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <span>
          StonkGen
        </span>
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="mr-auto navbar-nav"
        >
          <button
            className="btn btn-dark"
            data-test="home"
            onClick={[Function]}
          >
            Home
          </button>
          <li
            className="nav-item"
          >
            <button
              className="btn btn-dark"
              data-test="orders"
              onClick={[Function]}
            >
              Orders
            </button>
          </li>
          <li
            className="dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle nav-link"
              href="#"
              onClick={[Function]}
            >
              <span
                className="sr-only"
              >
                Toggle Dropdown
              </span>
            </a>
            <div
              aria-hidden={true}
              className="dropdown-menu dropdown-menu-right"
              role="menu"
              tabIndex="-1"
            >
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                History
              </button>
              <div
                className="dropdown-divider"
                onClick={[Function]}
                tabIndex="-1"
              />
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                Logout
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Hedaer Component should render the header snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <span>
          StonkGen
        </span>
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="mr-auto navbar-nav"
        >
          <button
            className="btn btn-dark"
            data-test="home"
            onClick={[Function]}
          >
            Home
          </button>
          <li
            className="nav-item"
          >
            <button
              className="btn btn-dark"
              data-test="orders"
              onClick={[Function]}
            >
              Orders
            </button>
          </li>
          <li
            className="dropdown nav-item"
            onKeyDown={[Function]}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle nav-link"
              href="#"
              onClick={[Function]}
            >
              <span
                className="sr-only"
              >
                Toggle Dropdown
              </span>
            </a>
            <div
              aria-hidden={true}
              className="dropdown-menu dropdown-menu-right"
              role="menu"
              tabIndex="-1"
            >
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                History
              </button>
              <div
                className="dropdown-divider"
                onClick={[Function]}
                tabIndex="-1"
              />
              <button
                className="dropdown-item"
                onClick={[Function]}
                role="menuitem"
                tabIndex="0"
                type="button"
              >
                Logout
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
